omod/src/main/webapp/fragments/field/dropDown.gsp

```diff
@@ -53,21 +53,73 @@
 <script type="text/javascript">
+
     var viewModel = viewModel || {};
     viewModel.validations = viewModel.validations || [];
-
     viewModel.${ config.id } = ko.observable();
-    <% if (required) { %>
-    viewModel.validations.push(function() {
-        return jq('#${ config.id }-field').is(':disabled') || (viewModel.${ config.id }() ? true : false);
-    });
-    <% } else { %>
-    viewModel.validations.push(function() {
-        viewModel.${ config.id }();
-        return true;
-    });
+
+    <% if (config.dependency || required) { %>
+
+        <% if (required) { %>
+        viewModel.validations.push(function() {
+            return jq('#${ config.id }-field').is(':disabled') || (viewModel.${ config.id }() ? true : false);
+        });
+        <% } else { %>
+        viewModel.validations.push(function() {
+            viewModel.${ config.id }();
+            return true;
+        });
+        <% } %>
     <% } %>
+
+    <% if (config.autocomplete) { %>
+        (function(jq) {
+            let selectListElement = jq("#${ config.id }-field");
+
+            // Create a new jQuery autocomplete text box, use it to update select list, which is hidden
+            let options = [];
+            selectListElement.find('option').each(function () {
+                let val = jq(this).val();
+                if (val !== '') {
+                    options.push({'label': jq(this).html(), 'value': val})
+                }
+            });
+            let inputBox = jq('<input type="text" class="dropdown-field-textbox" autocomplete="do-not-fill" data-lpignore="true">');
+            inputBox.autocomplete({
+                source: options,
+                minLength: 0,
+                matchContains: true,
+                select: function (event, ui) {
+                    event.preventDefault();
+                    inputBox.val(ui.item.label);
+                    selectListElement.val(ui.item.value);
+                    selectListElement.change();
+                },
+                focus: function (event, ui) {
+                    event.preventDefault();
+                    inputBox.val(ui.item.label);
+                },
+            }).blur(function () {
+                if (inputBox.val() !== selectListElement.find("option:selected").html()) {
+                    inputBox.val(null);
+                    selectListElement.val(null);
+                }
+            }).focus(function () {
+                jq(this).autocomplete('search', jq(this).val());
+            });
+            inputBox.insertAfter(selectListElement);
+            selectListElement.hide();
+        })(jQuery);
+    <% } %>
+
 </script>
-<% } %>
\ No newline at end of file
+
+<style>
+    #${ config.id }-field .ui-autocomplete {
+        max-height: 300px;
+        overflow-y: auto;   /* prevent horizontal scrollbar */
+        overflow-x: hidden; /* add padding to account for vertical scrollbar */
+        z-index:1000 !important;
+    }
+</style>
```

Comment:
@brandones - it's not immediately clear if this has changed since the version I committed.  The main thing I was concerned about is that when used in the one-question-per-screen workflow it wouldn't be obvious if an invalid value was entered when hitting tab or enter and going to the next question.  My intent was to add validation in place such that if text was entered and this ultimately didn't match the html of the select list element selected to mark the question in red/invalid in order to make clear that no value value was entered and the user should not proceed.

Reply:
Whoops, right, I didn't fix that use case; I only made it so that bad input text would automatically be deleted (which does block progress in the case that the field is required).